includes/gateways/stripe/includes/payment-methods/class-give-stripe-becs.php

```diff
@@ -0,0 +1,348 @@
+					$this->save_stripe_customer_id( $stripe_customer_id, $donation_id );
+					give_update_meta( $donation_id, '_give_stripe_customer_id', $stripe_customer_id );
+
+					// Save Source ID to donation note and DB.
+					give_insert_payment_note( $donation_id, 'Stripe Source/Payment Method ID: ' . $payment_method_id );
+					give_update_meta( $donation_id, '_give_stripe_source_id', $payment_method_id );
+					give_update_meta( $donation_id, '_give_stripe_payment_method_id', $payment_method_id );
+
+					// Save donation summary to donation.
+					give_update_meta( $donation_id, '_give_stripe_donation_summary', $donation_summary );
+
+					/**
+					 * This filter hook is used to update the payment intent arguments.
+					 *
+					 * @since 2.5.0
+					 */
+					$intent_args = apply_filters(
+						'give_stripe_create_intent_args',
+						array(
+							'amount'               => $this->format_amount( $donation_data['price'] ),
+							'currency'             => give_get_currency( $form_id ),
+							'payment_method_types' => [ 'au_becs_debit' ],
+							'statement_descriptor' => give_stripe_get_statement_descriptor(),
+							'description'          => give_payment_gateway_donation_summary( $donation_data ),
+							'metadata'             => $this->prepare_metadata( $donation_id ),
+							'customer'             => $stripe_customer_id,
+							'payment_method'       => $payment_method_id,
+							'confirm'              => true,
+							'setup_future_usage'   => 'off_session',
+							'mandate_data'         => [
+								'customer_acceptance' => [
+									'type'   => 'online',
+									'online' => [
+										'ip_address' => give_get_ip(),
+										'user_agent' => give_get_user_agent(),
+									],
+								],
+							],
+							'return_url'           => give_get_success_page_uri(),
+						)
+					);
+
+					// Send Stripe Receipt emails when enabled.
+					if ( give_is_setting_enabled( give_get_option( 'stripe_receipt_emails' ) ) ) {
+						$intent_args['receipt_email'] = $donation_data['user_email'];
+					}
+
+					$intent = $this->payment_intent->create( $intent_args );
+
+					if ( ! empty( $intent->status ) && 'processing' === $intent->status ) {
+
+						// Save Payment Intent Client Secret to donation note and DB.
+						give_insert_payment_note( $donation_id, 'Stripe Payment Intent Client Secret: ' . $intent->client_secret );
+						give_update_meta( $donation_id, '_give_stripe_payment_intent_client_secret', $intent->client_secret );
+
+						// Set Payment Intent ID as transaction ID for the donation.
+						give_set_payment_transaction_id( $donation_id, $intent->id );
+						give_insert_payment_note( $donation_id, 'Stripe Charge/Payment Intent ID: ' . $intent->id );
+
+						// Success. Send user to success page.
+						give_send_to_success_page();
+					} else {
+						give_send_back_to_checkout( '?payment-mode=' . give_clean( $_GET['payment-mode'] ) );
```

Comment:
I will prefer to use `$donation_data` to get value instead of directely reaching global constant